### PR TITLE
Smooth transition for Cards

### PR DESCRIPTION
**Pull request by vzamanillo**
Smooth transition to show movie or show cards when loading the respective views.

### CODE REVIEW

**Comment by amilajack** [comment]
If its not too much work, can you try showing me a screen recording of this? I want to make sure what it does before merging.

**Comment by vzamanillo** [comment]
Of course

smooth

File: app/styles/utilities/animation.scss [hunk 1]
```diff
@@ -1,8 +1,11 @@
-.fade {
-  opacity: 0;
-
-  .in {
+@keyframes in {
```
Comment (amilajack) [inline app/styles/utilities/animation.scss]: It seems like the indentation is a bit off here.

File: app/styles/components/CardList.scss [hunk 2]
```diff
@@ -45,7 +45,6 @@
       width: 100%;
       height: 219px;
       position: relative;
-      background-color: #D8D8D8;
```
Comment (amilajack) [inline app/styles/components/CardList.scss]: The initial reason why this was added was to show a poster as a placeholder for images. In other words, in the case of slow poster image loading, the background color would temporarily serve a placeholder. Maybe moving it to a darker shade would be a solution?

File: app/styles/utilities/animation.scss [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
+  from {
+    opacity: 0;
+  }
+  to {
```
Comment (amilajack) [inline app/styles/utilities/animation.scss]: These keyframes add a pretty nice effect. In my opinion, I really think we should with an effect similar to what [this website](https://devart.withgoogle.com/) has. Try clicking on a language and watch the list pull its old elements in and then push its new ones out. I think its pretty elegant. What do you think?

File: app/styles/components/CardList.scss (hunk 2 above)
Comment (vzamanillo) [inline app/styles/components/CardList.scss]: It's a perfect solution, what color do you think would be appropiated?

Comment (amilajack) [inline app/styles/components/CardList.scss]: I think we should keep the same color and change it when we get to release 0.0.6

File: app/styles/utilities/animation.scss (hunk 3 above)
Comment (vzamanillo) [inline app/styles/utilities/animation.scss]: Looks very nice, We can try it.

File: app/styles/components/CardList.scss (hunk 2 above)
Comment (vzamanillo) [inline app/styles/components/CardList.scss]: Ok, done!

**Comment by amilajack** [comment]
@vzamanillo [This line](https://travis-ci.org/amilajack/popcorn-time-desktop/jobs/154808996#L530) in the travis logs is failing because of stylings changes.

**Comment by amilajack** [comment]
I've just noticed that in the gif you uploaded, the text of the cards isn't part of the effect. Maybe adding the effect to a parent would make it work?

File: app/components/card/Card.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ export default class Card extends Component {
 
   render() {
     const placeholder =
-      '../../images/posterholder.png';
+      'images/posterholder.png';
```
Comment (amilajack) [inline app/components/card/Card.js]: Does the same path also work after being packaged? The paths in development and production a little different in some cases.

Comment (vzamanillo) [inline app/components/card/Card.js]: Sorry for the late answer, to package the app on linux 'icnsutils' and 'graphicsmagick' packages needs to be installed via

`sudo apt install graphicsmagick icnsutils`

I was getting the respective `ENOENT` errors.

Yes, it works, the image path is resolved without problems.

**Comment by amilajack** [comment]
Thanks for all your work! This is ready to be merged right?

**Comment by vzamanillo** [comment]
Hmmm, not at all, the transition effect occurs when the images are already cached, after the first run, at the first run time the card shows the background color while the image is loading, maybe needs to be improved with an image pre-loading...

**Comment by amilajack** [comment]
https://devart.withgoogle.com/

**Comment by ZakariaRidouh** [comment]
Ah, those animations look really good :smile:! 
I'll make a PR soon to bring the current design closer to the mock-up (logo in the middle).

**Comment by vzamanillo** [comment]
Something like this?

smooth

The capture is about 8 FPS, so the real effect is much better.

**Comment by vzamanillo** [comment]
Something strange is happening with the `test-e2e`, sometimes exceeds the timeout,

`Error: timeout of 10000ms exceeded.`

 if no timeout occurs the movieTitles are empty and I don't know why... 

Maybe is related to the selector?

**Comment by amilajack** [comment]
I dont think the items are being counted properly after the animation.

**Comment by amilajack** [comment]
Also the animation that i was thinking of was something like this:

popcorn-time-desktop

**Comment by vzamanillo** [comment]
Same test pass on Linux without problems... fails on Windows.

**Comment by amilajack** [comment]
Try manually testing the search functionality. If that works, then the PR is ready

**Comment by amilajack** [comment]
Can you try to get the animation to be something like this:

popcorn-time-desktop

**Comment by vzamanillo** [comment]
Of course, I will try to do it.

**Comment by amilajack** [comment]
@vzamanillo How'd the result turn out? If you couldn't do it that's fine. We'll do that in the 0.0.6 release (aka the design overhaul release).

**Comment by amilajack** [comment]
Let me know when this is ready to be merged

**Comment by vzamanillo** [comment]
I think the actual PR is ready to be merged, let's wait for the 0.06 to do your proposed transition, I have to do more tests and change some code to finish it. Thank you

**Comment by amilajack** [comment]
@vzamanillo sounds good! Merging.

**Comment by amilajack** [comment]
I have reverted this due to test failures. Let's reapply this merge when we start working on 0.0.6.

**Comment by amilajack** [comment]
Also i've been working on the effect:


transition_cards

**Comment by vzamanillo** [comment]
Cool! Looks good :)

**Comment by amilajack** [comment]
I've been working on the effects for this in another repo of mine:

```
hub clone amilajack/animation-experiments
cd animation-experiments
npm i
open staggered-cards/index.html
```

if you dont have ```hub```, ```brew install hub```. And if you're on windows...well...then you're on windows.

**Pull request merged**